### ISSUE TITLE
make install sed error

### ISSUE THREAD

**Issue by tfpoage** [johnhcc/homebrew-gs-x11#1]
Ran: brew install johnhcc/gs-x11/gv and got:

`...
==> make install
Last 15 lines from /Users/tom/Library/Logs/Homebrew/gv/02.make:
clang -DHAVE_CONFIG_H -I.  -DNON_BLOCKING_IO -DGV_CODE -DGV_LIBDIR=\"/usr/local/Cellar/gv/3.7.4/share/gv\" -I../lib -I../lib -g     -g -O2 -DXAW_INTERNATIONALIZATION -DXAW_MULTIPLANE_PIXMAPS -DXAW_GRAY_BLKWHT_STIPPLES -DXAW_ARROW_SCROLLBARS -I/usr/local/Cellar/libxau/1.0.9/include -I/usr/local/Cellar/libxdmcp/1.1.3/include -I/usr/local/Cellar/libxcb/1.14_1/include -I/usr/local/Cellar/libx11/1.7.0/include -I/usr/local/Cellar/libxext/1.3.4/include -I/usr/local/Cellar/libice/1.0.10/include -I/usr/local/Cellar/libsm/1.2.3/include -I/usr/local/Cellar/libxt/1.2.1/include -I/usr/local/Cellar/libxmu/1.1.3/include -I/usr/local/Cellar/libxaw3d/1.6.3/include -I/usr/local/Cellar/xorgproto/2021.3/include -c Scrollbar.c
creating gv_user.ad
creating gv_system.ad
creating GV.ad
cat ./gv-update-userconfig.in > gv-update-userconfig
creating GV
sed -e 's/^[^!]/!&/' GV.ad > ../GV
echo "__END__" >> gv-update-userconfig
sed: RE error: illegal byte sequence
cat GV.ad >> gv-update-userconfig
make[2]: *** [../GV] Error 1
make[2]: *** Waiting for unfinished jobs....
cat ./gv_misc_res.dat >> gv-update-userconfig
make[1]: *** [install] Error 2
make: *** [install-recursive] Error 1

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/johnhcc/homebrew-gs-x11/issues`

Noticed in ~/Library/Logs/Homebrew/gv/02.make:

`...
clang -DHAVE_CONFIG_H -I.  -DNON_BLOCKING_IO -DGV_CODE -DGV_LIBDIR=\"/usr/local/Cellar/gv/3.7.4/share/gv\" -I../lib -I../lib -g     -g -O2 -DXAW_INTERNATIONALIZATION -DXAW_MULTIPLANE_PIXMAPS -DXAW_GRAY_BLKWHT_STIPPLES -DXAW_ARROW_SCROLLBARS -I/usr/local/Cellar/libxau/1.0.9/include -I/usr/local/Cellar/libxdmcp/1.1.3/include -I/usr/local/Cellar/libxcb/1.14_1/include -I/usr/local/Cellar/libx11/1.7.0/include -I/usr/local/Cellar/libxext/1.3.4/include -I/usr/local/Cellar/libice/1.0.10/include -I/usr/local/Cellar/libsm/1.2.3/include -I/usr/local/Cellar/libxt/1.2.1/include -I/usr/local/Cellar/libxmu/1.1.3/include -I/usr/local/Cellar/libxaw3d/1.6.3/include -I/usr/local/Cellar/xorgproto/2021.3/include -c Scrollbar.c
creating gv_user.ad
creating gv_system.ad
creating GV.ad
cat ./gv-update-userconfig.in > gv-update-userconfig
creating GV
sed -e 's/^[^!]/!&/' GV.ad > ../GV
echo "__END__" >> gv-update-userconfig
sed: RE error: illegal byte sequence
cat GV.ad >> gv-update-userconfig
make[2]: *** [../GV] Error 1
make[2]: *** Waiting for unfinished jobs....
cat ./gv_misc_res.dat >> gv-update-userconfig
make[1]: *** [install] Error 2
make: *** [install-recursive] Error 1

HOMEBREW_VERSION: 3.0.10
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 0ad2564c5f1ddf6f16f707f2861556b80546713c
...`

Looks like Homebrew removed the build directory (or I can't find it). From the command line, sed has no problem with the regular expression. Configure issue or Makefile quoting? I'm happy to dig further.

Macos 11.2.3 x86

Thanks!

**Comment by tfpoage** [comment]
The problem may not be here, but with the combination of macOS BSD sed, NLS, and the contents of the GV.ad file. If I hand run configure and make, the BSD sed call likewise fails. Setting `LANG=C` allows the BSD sed command to succeed, and using Gnu sed (Homebrew) does not need the LANG override. My zsh environment happens to contain `LANG=en_US.UTF-8` (I don't recall setting this).

The `configure` script contains:
```
# NLS nuisances.
LC_ALL=C
export LC_ALL
LANGUAGE=C
export LANGUAGE
```
I'm not familiar with `LANGUAGE`, only `LANG`. These settings do not appear to make it into the Makefiles.

I see a few possible solutions:

1. Put `LC_ALL=C` on the `make install` command line in the `gv.rb` script here.
2. Add a dependency here on (Homebrew) gsed and edit `src/Makefile(.am)` here to use it.
3. Put the onus on Gnu to fix it.

Possibly relevant: https://unix.stackexchange.com/questions/509606/newlines-in-bsd-sed-vs-gsed

**Comment by tfpoage** [comment]
This results in a working gv:

`make install LC_ALL=C`

**Comment by johnhcc** [comment]
Thanks for looking into this!  I'll test your solution and make sure it doesn't break anything on the systems I have for testing, and make the change.  I much appreciate your report.

**Comment by johnhcc** [comment]
I'm still not 100% clear on why this is happening... but I've changed Line 21 in gv.rb from `system "make", "install"` to `system "make", "install", "LC_ALL=C"` for consistency with the `configure` script. At the very least this causes no problems for me, and at best it solves your problem! If you'll confirm that that's going to fix things on your end, I'll push the change.

Again, thanks for the report!

**Comment by tfpoage** [comment]
I traced the issue to a copyright line src/Makefile adds to some of the src/*.ad files while being composed, specifically the entry for José Marchesi (cf. below). The é is encoded in Latin-1 (ISO-8859-1; \351, 0xE9), and Apple's BSD sed in the Unicode (UTF-8) environment trips up on it (hence the LC_ALL=C override as a 'patch'). In nearly all other copyright lines in other files for this person, the 'e' does not have the accent (is ASCII). If I convert the Latin-1 é to UTF-8, Apple sed has no issue with it.

I initially considered this a bug with Apple's sed being overly picky, but now consider it an issue with Ghostview. It's old enough they may not want to fix it; I'll look into filing an issue with them. I installed gv only to have a second PostScript viewer while tracking down a music notation rendering bug in MuseScore 3.

../GV : GV.ad
        @echo "creating GV"
        sed -E -e 's/^[^!]/!&/' GV.ad > ../GV

GV.ad : gv_system.ad
        @echo "creating GV.ad"
        @cp gv_system.ad GV.ad

gv_system.ad : gv_user.ad
        @echo "creating gv_system.ad"
        @cp gv_user.ad gv_system.ad

gv_user.ad : $(srcdir)/gv_user_res.dat gv_intern_res_unix.dat gv_make_res.dat
        @echo "creating gv_user.ad"
        @echo "!"                                               >  gv_user.ad
        @echo "!  gv_user.ad"                           >> gv_user.ad
        @echo "!  User specific application defaults for gv"    >> gv_user.ad
        @echo "!  Copyright (C) 1995, 1996, 1997  Johannes Plass"       >> gv_user.ad
        @echo "!  Copyright (C) 2004,2005,2006,2007 José E. Marchesi"   >> gv_user.ad
        @echo "!"                                               >> gv_user.ad
        @cat $(srcdir)/gv_user_res.dat                                  >> gv_user.ad
        @cat gv_intern_res_unix.dat                             >> gv_user.ad
        @cat gv_make_res.dat                                    >> gv_user.ad

**Comment by tfpoage** [comment]
https://savannah.gnu.org/bugs/index.php?60337

**Comment by william8000** [comment]
I think that the files should spell people's names correctly, so the better solution is changing the makefile to use LC_ALL=C when running sed. Can you post a makefile that works? Or do you think that it is better to change the accented characters in all of the files from 8-bit Latin-1 ISO to UTF-8? I don't know what happened to the gv mainainer. I have a lot of gv patches, and I was thinking of posting an updated gv on github.

**Comment by johnhcc** [comment]
Interesting point, @william8000. It looks to me like the `make install LC_ALL=C` work-around produces a binary that is identical to that produced by systems with gnu sed installed (which contains the spellings without accented characters). I suppose either the authors weren't concerned about this, or weren't aware (or both).

I'll mess around with this a bit and see what I can come up with.  (If either of you comes up with something better first, please just update this post.)

Good question about what happened to maintenance of gv. If you decide to post an updated fork on github, let me know - it's easy enough for me to point this tap there.  Thanks!

**Comment by william8000** [comment]
I posted my repository to https://github.com/william8000/gv
I added LC_ALL=C before the `sed` line.
I am not sure what to do with the accented characters. I think that the old X11 code expects 8 bit ISO, but gv/src/nls has UTF-8, so maybe UTF-8 would work.
gv/NEWS has a list of my changes. Unfortunately it is all in one commit, but if you think that it would be useful but don't like some of the changes (in particular, I flipped pan and zoom on the mouse buttons), I could move those changes to a branch.
